File: Programmers/위장.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> map;
    for (vector<string> cloth : clothes)
        map[cloth[1]]++;

    for (auto it : map)
        answer *= (it.second + 1);
    
    return answer - 1;
}

File: Programmers/위장.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> map = new HashMap<>();
        for (String[] cloth : clothes)
            map.put(cloth[1], map.getOrDefault(cloth[1], 0) + 1);
        
        for (int v : map.values()) 
            answer *= (v + 1);
        return answer - 1;
    }
}